diff --git a/hw/i386/amd_iommu.c b/hw/i386/amd_iommu.c
index 2570a4701c..a9391bdee7 100644
--- a/hw/i386/amd_iommu.c
+++ b/hw/i386/amd_iommu.c
@@ -662,6 +662,7 @@ static uint64_t large_pte_page_size(uint64_t pte)
 static uint64_t fetch_pte(AMDVIAddressSpace *as, hwaddr address, uint64_t dte,
                           uint64_t *pte, hwaddr *page_size)
 {
+    printf("translate 0x%016"PRIx64"\n", address);
     IOMMUAccessFlags perms = amdvi_get_perms(dte);
 
     uint8_t level, mode;
@@ -691,11 +692,11 @@ static uint64_t fetch_pte(AMDVIAddressSpace *as, hwaddr address, uint64_t dte,
         return -AMDVI_FR_PT_ROOT_INV;
     }
 
+    printf("PT root (level=%d):  pte=%016"PRIx64"\n", level, *pte);
     do {
         level -= 1;
 
-        /* Update the page_size */
-        *page_size = PTE_LEVEL_PAGE_SIZE(level);
+        printf("PT walk (level=%d):  pte=%016"PRIx64" NextLevel=%llu page_size=0x%"PRIx64"\n", level, *pte, PTE_NEXT_LEVEL(*pte), *page_size);
 
         /* Permission bits are ANDed at every level, including the DTE */
         perms &= amdvi_get_perms(*pte);
@@ -714,6 +715,9 @@ static uint64_t fetch_pte(AMDVIAddressSpace *as, hwaddr address, uint64_t dte,
             break;
         }
 
+        /* Update the page_size */
+        *page_size = PTE_LEVEL_PAGE_SIZE(level);
+
         /*
          * Index the pgtable using the IOVA bits corresponding to current level
          * and walk down to the lower level.
@@ -746,10 +750,13 @@ static uint64_t fetch_pte(AMDVIAddressSpace *as, hwaddr address, uint64_t dte,
      * must be the first PTE of the series.
      */
     if (PTE_NEXT_LEVEL(*pte) == 7) {
+        printf("PTE NextLevel=7, doubling page size\n");
         /* Update page_size with the large PTE page size */
         *page_size = large_pte_page_size(*pte);
     }
 
+    printf("PTE found:  pte=%016"PRIx64" NextLevel=%llu page_size=0x%"PRIx64"\n", *pte, PTE_NEXT_LEVEL(*pte), *page_size);
+
     return 0;
 }
 
diff --git a/hw/misc/edu.c b/hw/misc/edu.c
index cece633e11..e9920f6c30 100644
--- a/hw/misc/edu.c
+++ b/hw/misc/edu.c
@@ -146,6 +146,8 @@ static void edu_dma_timer(void *opaque)
         return;
     }
 
+    printf("edu DMA: 0x%lx <-> 0x%lx\n", edu->dma.src, edu->dma.dst);
+
     if (EDU_DMA_DIR(edu->dma.cmd) == EDU_DMA_FROM_PCI) {
         uint64_t dst = edu->dma.dst;
         edu_check_range(dst, edu->dma.cnt, DMA_START, DMA_SIZE);
@@ -160,6 +162,8 @@ static void edu_dma_timer(void *opaque)
                 edu->dma_buf + src, edu->dma.cnt);
     }
 
+    printf("dma_buf = %s\n", edu->dma_buf);
+
     edu->dma.cmd &= ~EDU_DMA_RUN;
     if (edu->dma.cmd & EDU_DMA_IRQ) {
         raise_irq = true;
